File: comu/com.energyict.mdc.protocol.api/src/main/java/com/energyict/mdc/protocol/api/security/SecuritySuite.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.energyict.mdc.protocol.api.security;

import java.io.Serializable;
import java.util.List;

public interface SecuritySuite extends Serializable, DeviceAccessLevel {

    /**
     * A list of {@link EncryptionDeviceAccessLevel}s supported by this security suite
     */
    List<EncryptionDeviceAccessLevel> getEncryptionAccessLevels();

    /**
     * A list of {@link AuthenticationDeviceAccessLevel}s supported by this security suite
     */
    List<AuthenticationDeviceAccessLevel> getAuthenticationAccessLevels();

    /**
     * A list of all supported security levels for requests
     */
    List<RequestSecurityLevel> getRequestSecurityLevels();

    /**
     * A list of all supported security levels for responses
     */
    List<ResponseSecurityLevel> getResponseSecurityLevels();

}

File: comu/com.energyict.comserver.comserver.core.extjs/src/main/web/js/mdc/src/model/SecuritySuite.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Mdc.model.SecuritySuite', {
    extend: 'Ext.data.Model',
    fields: [
        {name: 'id',type:'number',useNull:true},
        {name: 'name', type: 'string', useNull: true}
    ]
});